assets/js/api/request.js

```
@@ -0,0 +1,46 @@
+import axios from 'axios';
+
+const url = new URL('http://localhost:4000/');
+
+export default {
+  async fetchData(config) {
+    let data;
+    url.pathname = config.pathname;
+
+    await axios.get(url.href).then((response) => {
+      const result = JSON.parse(JSON.stringify(response.data));
+
+      if (config.key) {
+        data = result[config.key];
+      } else {
+        data = result;
+      }
+    })
+      .catch(() => {
```

you are using an empty catch block here to handle errors, which means that errors will occur silently - this can lead to difficult to trace bugs/errors!

You can either handle the error by doing something, e.g. logging the error and rethrowing the error, or you can remove the catch block entirely to let the client's (components) handle the error instead. E.g. in app_bar.vue, I can do:

`
 try {
 await request.makeChanges(config);
 } catch (e) {
 // show some error message in the front-end
 }
`

---

ありがとうございます！